Release 0.9.0

blakeblackshear opened the pull request:
Changes:
- Recordings UI
- JSMpeg camera streams
- Birdseye camera view (closes #338 #799)
- Remove tmpfs_cache_size setting as it is no longer needed for hassos
- Delete events from the UI
- Code cleanup and refactors for best practices
- Convert config to pydantic to support future config editor
- Support for self signed certs and key auth with MQTT
- `clip_ready` event type published to events topic when clip has been saved
- `snapshot_time` added to events mqtt topic to determine if an updated snapshot is available
- Clean PNG snapshots alongside jpg for future training purposes
- Ability to customize timestamp drawn on snapshots and the debug video feed
- Dynamic motion detection defaults based on resolution (closes #1176)
- Adds option to specify jpeg quality on various http endpoints and in config (fixes #821)
- Trigger a restart from WebUI or via MQTT
- Labelmap can now be customized via config (fixes #507)
- Clips and recordings are merged and clips are now dynamically created based on recordings

Fixes #937
Fixes #920 
Fixes #799 
Fixes #1162 
Fixes #581 
Fixes #1231 
Fixes #1310

blakeblackshear committed:
add jsmpeg relay

blakeblackshear committed:
adding s6-overlay and running jsmpeg

herostrat commented on docker-compose.yml:
```diff
@@ -0,0 +1,37 @@
+      - /dev/dri:/dev/dri # for intel hwaccel, needs to be updated for your hardware
+    volumes:
+      - /etc/localtime:/etc/localtime:ro
+      - ./config/config.yml:/config/config.yml:ro
```
This line is wrong imho, or the docu is not clear to me
With this it expects a folder "config/config.yml" not a YAML file
My workaround is:

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -22,7 +22,7 @@ services:
       - /dev/dri:/dev/dri # for intel hwaccel, needs to be updated for your hardware
     volumes:
       - /etc/localtime:/etc/localtime:ro
-      - ./config/config.yml:/config/config.yml:ro
+      - ./config/config.yml/config.yml:/config/config.yml:ro
       - ./debug:/media/frigate
       - ./frigate:/opt/frigate/frigate:cached
       - ./migrations:/opt/frigate/migrations:cached

and adding the config.yml in the folder config/config.yaml/

herostrat committed:
Add draw_timstamp: Helper for timestamp mods

herostrat committed:
Use timestamp decorator helper in codebase

blakeblackshear committed:
wip yuv_rgb refactor

mitchross commented:
@blakeblackshear 

Whats plan to merge this into master? Anything I can help with?

blakeblackshear commented:
There are several other things I want to include in this release, so it will be several weeks. I haven't defined a finish line for this release yet.

jakekeeys commented on frigate/events.py:
```diff
@@ -33,162 +36,192 @@ def __init__(self, config, camera_processes, event_queue, event_processed_queue,
-        
+        # if the earliest event is more tha max seconds ago, cap it
+        earliest_event = max(
+            earliest_event, datetime.datetime.now().timestamp() - max_seconds
```
```suggestion
            earliest_event, datetime.datetime.now().timestamp() - self.config.clips.max_seconds
```

blakeblackshear committed:
basic plumbing for birdseye view

blakeblackshear committed:
updated devcontainer

blakeblackshear committed:
development tweaks

flacjacket committed:
Updates to object processing

Lock updates to tracked objects, current frame time, motion boxes, and
regions on `update()`.

Directly create Counters using counted values.

Don't convert removed_ids, new_ids, or updated_ids sets to lists.

Update defaultdict's to remove un-necessary lambdas when possible.

When possible, drop un-necessay list comprehensions, such as when
calling `any`.

Use set comprehension, rather than passing a list comprehension into
`set()`.

Do the slightly more pythonic `x not in y` rather than `not x in y` to
check list inclusion.

hunterjm committed:
Initial Recordings UI

hunterjm committed:
lint fixes

blakeblackshear committed:
adding output process to handle downstream processing of frames

blakeblackshear committed:
handle exit signals

blakeblackshear committed:
remove node jsmpeg server

blakeblackshear committed:
ensure output exits properly

blakeblackshear committed:
increase read size and listen on local

blakeblackshear committed:
only send frames to ffmpeg if there are clients listening

blakeblackshear committed:
manage multiple cameras

blakeblackshear committed:
removing old node relay

blakeblackshear committed:
move birdseye placeholder to output process

hunterjm committed:
video overlay

hunterjm committed:
add event card to overlay

hunterjm committed:
fix videojs bug when switching cameras, support recording delay, fix navigation highlight

hunterjm committed:
restyle to match Material Design List specs

hunterjm committed:
lint cleanup

hunterjm committed:
remove now playing because I do not handle autoadvance properly yet

hunterjm committed:
do not show event end time

hunterjm committed:
lint fix... again

hunterjm committed:
cleanup dead code

hunterjm committed:
We need to use relative URLs for Ingress to work

hunterjm committed:
recordings is taken by nginx so refresh fails - change base to recording

hunterjm committed:
add playback rate adjustment and seek buttons

hunterjm committed:
change seek back to 10 seconds

blakeblackshear committed:
add layout calculations

hunterjm committed:
add mobile player and fullscreen support for iOS

blakeblackshear committed:
Merge branch 'master' into release-0.9.0

Flamingi committed:
Added support for authentication with client certificate with MQTT broker

hunterjm committed:
hide recordings page if record is not enabled, show error if no recordings available.

hunterjm committed:
add/fix tests

hunterjm committed:
add sidebar test

hunterjm committed:
Update web/src/__tests__/Sidebar.test.jsx

hunterjm committed:
initial implementation of recordings table

hunterjm committed:
clean up VOD endpoint

hunterjm committed:
add default to reducer

hunterjm committed:
fetch recordings until end of hour, not top of next hour

hunterjm committed:
remove backfill - only store rows moving forward

blakeblackshear committed:
update birdseye layout calculations

blakeblackshear committed:
fix to birdseye connection

firefly2442 committed:
Update http.py

comparison of constants, use T/F instead

blakeblackshear committed:
use quality value and faster resize

hunterjm committed:
Cleanup video player and use consistently across recordings and events.

hunterjm committed:
fix lint and tests

hunterjm committed:
more lint fixes

blakeblackshear committed:
add config for birdseye

blakeblackshear committed:
use values from config

blakeblackshear committed:
implement birdseye modes

blakeblackshear committed:
use actual logger

blakeblackshear committed:
precompute the layout offsets and dimensions

blakeblackshear committed:
cap the array size of the output queue

blakeblackshear committed:
set jsmpeg video size to same as camera

blakeblackshear committed:
removing testing files

blakeblackshear committed:
updating docs

blakeblackshear committed:
functioning birdseye view

blakeblackshear committed:
disable audio

blakeblackshear committed:
lock before iterating web sockets

blakeblackshear committed:
output the current frame from each active camera instead of last active

blakeblackshear committed:
trickle in some frames after activity stops

herostrat commented on docker-compose.yml:
```diff
@@ -11,18 +12,15 @@ services:
       - "5000:5000"
       - "5001:5001"
       - "8080:8080"
+    entrypoint: ["sudo", "/init"]
```
FYI this produces a error for me and I cannot start the frigate-dev container.
```
sudo: /init: command not found
```

blakeblackshear replied:
Make sure you rebuild frigate:latest

herostrat replied:
Thanks, that was my problem. Sorry for the noise.

blakeblackshear committed:
overlay the frigate logo on the blank frame

blakeblackshear committed:
swith camera view to jsmpeg

blakeblackshear committed:
lint fixes

blakeblackshear committed:
fix some test errors

blakeblackshear committed:
remove gevent fixes #920

blakeblackshear committed:
cleanup

blakeblackshear committed:
send clip_ready mqtt to event topic

herostrat committed:
Add and use config for timestamp style

herostrat committed:
Add doc about timestamp styles

herostrat committed:
Do apt update before upgrade

hunterjm committed:
fix tests

blakeblackshear committed:
Update cameras.md

gpete committed:
Fixed overwritten argument 'media' (#1026)

media variable name is reused and overwritten causing issues with event expiration & clean up.
Also changed name in pure_duplicates for consistency.

mrdrup committed:
Fix 'FileExistsError' shared memory exception (#945)

blakeblackshear committed:
adding clean_copy to snapshot config

blakeblackshear committed:
save clean snapshot

blakeblackshear committed:
add snapshot time to event data

blakeblackshear committed:
cleanup clean snapshots

blakeblackshear committed:
update docs

blakeblackshear committed:
add fullscreen to jsmpeg player

blakeblackshear committed:
remove console log

Commit:
Fix review finding

ElMoribond committed:
Add ability to restart

ElMoribond committed:
Add ability to restart

ElMoribond committed:
Add ability to restart

ElMoribond committed:
Add ability to restart

ElMoribond committed:
Add ability to restart

ElMoribond committed:
Add ability to restart

blakeblackshear committed:
reset layout_dim too

blakeblackshear committed:
no need to hang onto the last active frame

blakeblackshear committed:
fixup timestamp config

ElMoribond committed:
Add ability to restart

blakeblackshear committed:
maintain aspect ratio for birdseye

blakeblackshear committed:
clear position before copy

ElMoribond committed:
Add ability to restart

ElMoribond committed:
Add ability to restart

ElMoribond committed:
Add ability to restart

blakeblackshear committed:
add config for camera live stream

blakeblackshear committed:
expand dockerignore

ElMoribond committed:
Add ability to restart

ElMoribond committed:
Add ability to restart

I restored the original line as it can be misleading.

hunterjm committed:
initial conversion to pydantic

hunterjm committed:
remove unneeded changes

ElMoribond committed:
Add ability to restart

disable warning

ElMoribond committed:
Add ability to restart

ElMoribond committed:
Add ability to restart

hunterjm committed:
fix ffmpeg config and remove side effects

hunterjm committed:
remove side effects from password substitution

hunterjm committed:
fix timestamp

firefly2442 committed:
identity check is correct way to check for None

blakeblackshear committed:
make max_disappeared optional with default value

blakeblackshear committed:
add test for max_disappeared default

blakeblackshear committed:
formatting

blakeblackshear committed:
no-cache and pull for build

blakeblackshear committed:
update to 4.3.2

blakeblackshear committed:
use non-free va driver fixes #581

ccitro committed:
Prevent repeated db connect calls

Only call database.connect() if database.is_closed()
Prevents errors if _db_connect is triggered twice

ccitro committed:
Check for jpg bytes before make_response

If jpg_bytes wasn't retrieved from either desk or a tracked object, respond with 404
Prevents uncaught error for unknown event ids sent to event_snapshot endpoint

blakeblackshear committed:
fix cache cleanup (fixes #1231)

blakeblackshear committed:
set min motion frame to 180

blakeblackshear committed:
make contour_area dynamic

blakeblackshear committed:
update motion docs (fixes #1176)

blakeblackshear committed:
add options to define jpeg quality

blakeblackshear committed:
add range to birdseye quality config

sinamics committed:
resolve #1143

blakeblackshear committed:
use intel repo for va drivers

blakeblackshear committed:
no-cache for docker buiilds

blakeblackshear committed:
update makefile versions

blakeblackshear committed:
use kisak-mesa repo

sinamics committed:
error handling and linting

sinamics committed:
set top postion to widow_padding

blakeblackshear committed:
no need to worry about payload here

blakeblackshear committed:
remove default value changes

blakeblackshear committed:
update docs

blakeblackshear committed:
useCallback here too

blakeblackshear committed:
publish on zone change (fixes #1310)

blakeblackshear committed:
allow specific objects to be listed for a zone (fixes #1123)

blakeblackshear committed:
make default motion a little more sensitive

blakeblackshear committed:
fix test

blakeblackshear committed:
add labelmap customization to the config (fixes #507)

blakeblackshear committed:
revert ffmpeg version

blakeblackshear committed:
arch specific s6-overlay

hunterjm committed:
initial commit

hunterjm committed:
fix recording retention logic

blakeblackshear committed:
ensure default args propogate

blakeblackshear committed:
include ffmpeg_cmds in config endpoint

hunterjm committed:
update docs

hunterjm committed:
optimize recording maintenance logic

hunterjm committed:
break out recording maintenance and cleanup into separate threads

hunterjm committed:
change MQTT to toggle recordings instead of clips

hunterjm committed:
bump VOD max clips

hunterjm committed:
add download option on clips and snapshots

hunterjm committed:
fix DoesNotExist exception

hunterjm committed:
remove -f mp4 as it is not needed

hunterjm committed:
add secure token module to NGINX in order to pass authSig down to segment files

blakeblackshear committed:
fix motion mask overlay

ElMoribond committed:
add tabbed buttons component

ElMoribond committed:
add tabbed buttons component

blakeblackshear committed:
handle exception when missing frame

blakeblackshear committed:
switch default live resolution to 720

blakeblackshear committed:
increase jsmpeg buffer size

blakeblackshear committed:
update for coral updates

blakeblackshear committed:
use new pycoral libraries

blakeblackshear committed:
fix scaling

blakeblackshear committed:
use a nginx internal redirect

blakeblackshear committed:
cleanup dynamic mp4 clips

blakeblackshear committed:
assume the clip for the event exists with recordings

blakeblackshear committed:
clarify a few things in logs

blakeblackshear committed:
move width/height/fps under detect and make required

also resizes the output from ffmpeg to specified size

blakeblackshear committed:
Merge remote-tracking branch 'origin/master' into release-0.9.0

blakeblackshear committed:
update FAQ

blakeblackshear committed:
clean house on clips

blakeblackshear committed:
stop using pycoral libs for efficiency

herostrat committed:
Fix logger invocation

blakeblackshear committed:
cleanup logging

blakeblackshear committed:
use CPu detector by default

blakeblackshear committed:
move colormap to config

blakeblackshear committed:
copy then delete (fixes #1516)

Commit:
removed comma. This was causing the main window to be pulled down behind the headerbar, hence the odd menu behavior

Commit:
reduced navbar padding / height

blakeblackshear committed:
use find to reduce CPU usage for legacy expiration

blakeblackshear committed:
limit legacy expiration to files after the oldest recording in the db

onedr0p committed:
fix: consistent error logging to mqtt connection issues (#1578)

sinamics committed:
Move event-view to events table. (#1596)

* fixed position for Dialog

* added eventId to deleted item

* removed page route redirect + New Close Button

* event component added to events list. New delete reducer

* removed event route

* moved delete reducer to event page

* removed redundant event details

* keep aspect ratio

* keep aspect ratio

* removed old buttons - repositioned to top

* removed console.log

* event view function

* removed clip header

* top position

* centered image if no clips avail

* comments

* linting

* lint

* added scrollIntoView when event has been mounted

* added Clip header

* added scrollIntoView to test

* lint

* useRef to scroll event into view

* removed unused functions

* reverted changes to event.test

* scroll into view

* moved delete reducer

* removed commented code

* styling

* moved close button to right side

* Added new close svg icon

Co-authored-by: Bernt Christian Egeland <cbegelan@gmail.com>

blakeblackshear committed:
reverse sort within a day for recordings

blakeblackshear committed:
fix match for websocket url (fixes #1633)

blakeblackshear committed:
handle missing camera names

blakeblackshear committed:
use model from frogfish release

blakeblackshear committed:
error on invalid role

blakeblackshear committed:
cleanup global and camera detect config (fixes #1615)

blakeblackshear committed:
sync global snapshot options (fixes #1621)

blakeblackshear committed:
tweak vod settings for varying iframe intervals

blakeblackshear committed:
reverse sort events within hour

blakeblackshear committed:
properly handle scenario with no recordings

Commit:
hide birdseye nav if not enabled

blakeblackshear committed:
reduce db queries for recording cleanup

blakeblackshear committed:
dont expire events in two places

blakeblackshear committed:
optimize checking recordings for events

sorts events and recordings so you can avoid a cartesian product of checking all events against all recordings

blakeblackshear committed:
bulk delete recordings

blakeblackshear committed:
moar speed

drinfernoo committed:
Add RTMP and timestamp style to global config (#1674)

* :memo::white_check_mark::wrench: - Make RTMP config global

Fixes #1671

* :memo::white_check_mark::wrench: - Make timestamp style config global

Fixes #1656

* fix test function names

* formatter

Co-authored-by: Blake Blackshear <blakeb@blakeshome.com>

sinamics committed:
Events performance (#1645)

* rearrange event route and splitted into several components

* useIntersectionObserver

* re-arrange

* searchstring improvement

* added xs tailwind breakpoint

* useOuterClick hook

* cleaned up

* removed some video controls for mobile devices

* lint

* moved hooks to global folder

* moved buttons for small devices

* added button groups

Co-authored-by: Bernt Christian Egeland <cbegelan@gmail.com>

blakeblackshear committed:
aarch64 makefile fix

blakeblackshear committed:
remove font_scale in timestamp_style and calculate dynamically again

blakeblackshear committed:
fix color config for ts (fixes #1679)

blakeblackshear committed:
disallow extra keys in config

blakeblackshear committed:
remove nested enabled config setting on events

blakeblackshear committed:
Merge remote-tracking branch 'origin/master' into release-0.9.0

blakeblackshear committed:
store start/end event with pre/post capture to avoid expiring wanted recordings

hunterjm committed:
allow for custom object detection model via configuration

blakeblackshear committed:
docs updates

blakeblackshear committed:
remove fps arg for mjpeg

blakeblackshear committed:
fix global live config

blakeblackshear committed:
cleanup ha notification docs

blakeblackshear committed:
add version to the logs on startup

blakeblackshear committed:
more docs updates

blakeblackshear committed:
remove clip_ready event type

this doesnt really mean anything more than "end" anymore. new has_clip property added

blakeblackshear committed:
add udp camera example to docs

blakeblackshear committed:
make expiration of deleted camera footage faster

blakeblackshear committed:
simplify logic and fix wrong segments expiring (fixes #1779)

blakeblackshear committed:
docs clarification for masks

blakeblackshear committed:
use s6 to shutdown frigate

blakeblackshear committed:
allow partial days in retention settings

blakeblackshear committed:
run nginx as root

this addresses an issue many have had when using network shares

blakeblackshear committed:
add region/bbox/area to event table

kpelzel committed:
change theme color from red

blakeblackshear committed:
install docs clarification

blakeblackshear committed:
add support for rockchip hwaccel

blakeblackshear committed:
instantiate area field

blakeblackshear committed:
docs config update for netlify

blakeblackshear committed:
fix path

blakeblackshear committed:
add affiliate links

blakeblackshear committed:
update docs url

blakeblackshear committed:
docs updates

blakeblackshear committed:
better log message when corrupt segment detected

blakeblackshear committed:
delete clean snapshot when duplicate

blakeblackshear committed:
revert rockchip support for aarch64

blakeblackshear committed:
optimize images for web

blakeblackshear committed:
update latest recommendations for reolink

blakeblackshear committed:
updated links

blakeblackshear committed:
further doc clarifications

blakeblackshear committed:
update hardware recommendations

kpelzel committed:
added white background to apple-touch-icon
